lib/wallet/api/http/Cardano/Wallet/Api/Http/Shelley/Server.hs

```diff
@@ -4410,56 +4428,22 @@ mkApiTransaction timeInterpreter wrk wid timeRefLens tx = do
             $ W.shelleyOnlyReadRewardAccount @s @k @n db wid
         pure $ mkApiAnyCertificate rewardAccount derivPath <$> certificates
 
-    depositIfAny :: Natural
-    depositIfAny
-        | tx ^. (#txMeta . #direction) == W.Outgoing =
-            if totalIn < totalOut
-            then 0
-            else totalIn - totalOut
-        | otherwise = 0
-
-    -- (pending) when reclaim is coming we have (fee+out) - inp = deposit
-    -- tx is incoming, and the wallet spent for fee and received deposit - fee as out
-    -- (inLedger) when reclaim is accommodated we have out - inp < deposit as fee was incurred
-    -- So in order to detect this we need to have
-    -- 1. deposit
-    -- 2. have inpsWithoutFee of the wallet non-empty
-    -- 3. outs of the wallet non-empty
-    -- 4. tx Incoming
-    -- 5. outs - inpsWithoutFee <= deposit
-    -- assumption: this should work when
-    depositValue :: Natural
-    depositValue = fromIntegral . unCoin $ tx ^. #txDeposit
-
-    reclaimIfAny :: Natural
-    reclaimIfAny
-        | tx ^. (#txMeta . #direction) == W.Incoming =
-              if ( totalIn > 0 && totalOut > 0 && totalOut > totalIn)
-                 && (totalOut - totalIn <= depositValue) then
-                  depositValue
-              else
-                  0
-        | otherwise = 0
-
-    totalIn :: Natural
-    totalIn
-        = sum (txOutValue <$> mapMaybe snd (tx ^. #txInputs))
-        + sum (fromIntegral . unCoin <$> Map.elems (tx ^. #txWithdrawals))
-
-    totalOut :: Natural
-    totalOut
-        = sum (txOutValue <$> tx ^. #txOutputs)
-        + maybe 0 (fromIntegral . unCoin) (tx ^. #txFee)
-
-    txOutValue :: TxOut -> Natural
-    txOutValue = fromIntegral . unCoin . TxOut.coin
-
     toAddressAmountNoAssets
         :: TxOut
         -> AddressAmountNoAssets (ApiT Address, Proxy n)
     toAddressAmountNoAssets (TxOut addr (TokenBundle.TokenBundle coin _)) =
         AddressAmountNoAssets (ApiT addr, Proxy @n) (mkApiCoin coin)
 
+    depositEventTaken :: Tx.DepositEvent -> Quantity "lovelace" Natural
```

Comment:
what about relocating those functions near DepositEvent definition? So in Cardano.Wallet.Transaction module?

Reply:
I have intentionally left these functions in the API layer, as IMO they should not ideally exist at all, and the only reason they are is that API layer renders those events as `Quanity "lovelace" Natural`.

Reply:
fair enough. small nitpick - Quantity type is not exactly Api level type, at least when it comes to location. It just enriches "content" and make the same content differentiable. Of course it has nice API level application.